retrieving/retrieving-and-querying-mongodb-documents-with-go.md

```diff
@@ -47,108 +47,117 @@ Knowing the fields that exist in our data will be important to us when it comes
 
 ## Reading All Documents from a Collection
 
-Reading all data from a collection consists of making the request, then looping through the results. Knowing what fields exist on each of the documents isn't too important, only knowing the collection name itself.
+Reading all data from a collection consists of making the request, then working with the results cursor. Knowing what fields exist on each of the documents isn't too important, only knowing the collection name itself.
 
 A simple example of this can be done through the following:
 
-```golang
+```go
 cursor, err := episodesCollection.Find(ctx, bson.M{})
 if err != nil {
     log.Fatal(err)
 }
-for cursor.Next(ctx) {
-    var episode bson.D
-    cursor.Decode(&episode)
-    fmt.Println(episode)
-}
-cursor.Close(ctx)
+var episodes []bson.M
+cursor.All(ctx, &episodes)
```

Comment:
The error should be checked from `All` (here and anywhere else). I recommend running `errcheck` on any code you write to make sure errors aren't being swallowed (https://github.com/kisielk/errcheck).

Reply:
Thanks for the suggestion on that package. Wasn't aware of it, but can already tell it is going to be awesome!